# HG changeset patch
# User Betterbird <betterbird@betterbird.eu>
# Date 1704994698 -3600
# Parent  048126f7db45a08ee49fee718e7f046c3c0f6662
NNN6 - Wait until image has a width before setting 'overflowing' attribute.

diff --git a/mail/base/content/aboutMessage.js b/mail/base/content/aboutMessage.js
--- a/mail/base/content/aboutMessage.js
+++ b/mail/base/content/aboutMessage.js
@@ -39,24 +39,28 @@ const messengerBundle = Services.strings
 
 var gMessage, gMessageURI;
 var autodetectCharset;
 
 function getMessagePaneBrowser() {
   return document.getElementById("messagepane");
 }
 
-function messagePaneOnResize() {
+async function messagePaneOnResize() {
   const doc = getMessagePaneBrowser().contentDocument;
   // Bail out if it's http content or we don't have images.
   if (doc?.URL.startsWith("http") || !doc?.images) {
     return;
   }
 
   for (let img of doc.images) {
+    let count = 0;
+    while (count++ < 5 && img.clientWidth == 0) {
+      await new Promise(resolve => setTimeout(resolve, 100));
+    }
     img.toggleAttribute(
       "overflowing",
       img.clientWidth - doc.body.offsetWidth >= 0 &&
         (img.clientWidth <= img.naturalWidth || !img.naturalWidth)
     );
   }
 }
 
